### PR TITLE
Update Jump Code actions

### PR DESCRIPTION
**Pull request by PizieDust**
This PR aims to fix some of the issues related to the Jump code actions. 
- [x] Ability to add specific key-bindings for each jump. [Original comment](https://github.com/ocaml/ocaml-lsp/pull/1364#issuecomment-2352293536)
- [x] Toggling on and off the display of the jump targets on context menus, (`ON` by default) [Original comment, point 1](https://github.com/ocaml/ocaml-lsp/pull/1364#issuecomment-2353283155)
- [x] Navigating to a specific jump on the context menu using the keyboard (pressing the first letter of the jump target) [Original comment](https://github.com/ocaml/ocaml-lsp/pull/1364#issuecomment-2350224747)

cc @voodoos @xvw @rgrinberg @awilliambauer

### CODE REVIEW

**Comment by PizieDust** [comment]
@xvw Please add the `no-changelog` label.

File: ocaml-lsp-server/test/e2e-new/code_actions.ml [hunk 1]
```diff
@@ -1288,25 +1288,10 @@ let f (x : t) (d : bool) =
     let end_ = Position.create ~line:5 ~character:5 in
     Range.create ~start ~end_
   in
-  print_code_actions source range ~filter:(find_action "merlin-jump");
+  print_code_actions source range ~filter:(find_action "merlin-jump-match");
```
Comment (voodoos) [inline ocaml-lsp-server/test/e2e-new/code_actions.ml:1291]: Now that only one precise jump is printed, could you expand the test to show that the jumps to let/fun/next-case are still there ?

**Comment by voodoos** [comment]
> @xvw Please add the `no-changelog` label.

Since we do add usef-facing changes (a new configuration option) we should add it to the changelog :-)

File: ocaml-lsp-server/docs/ocamllsp/config.md [hunk 2]
```diff
@@ -42,5 +42,12 @@ interface config {
   * @since 1.18
   */
   syntaxDocumentation: { enable : boolean }
+
+  /**
+  * Enable/Disable Merlin Jump code actions
+  * @default true
+  * @since 1.19
+  */
+  merlinJumpsCodeAction: { enable : boolean }
```
Comment (voodoos) [inline ocaml-lsp-server/docs/ocamllsp/config.md:51]: This doesn't match the actual name of the config option: `merlinJumpCodeActions`

File: ocaml-lsp-server/test/e2e-new/code_actions.ml [hunk 3]
```diff
@@ -1297,31 +1297,114 @@ let f (x : t) (d : bool) =
...
           {
-            "end": { "character": 2, "line": 4 },
-            "start": { "character": 2, "line": 4 }
+            "end": { "character": 3, "line": 6 },
+            "start": { "character": 3, "line": 6 }
           }
         ],
         "command": "ocamllsp/merlin-jump-to-target",
-        "title": "Jump to match"
+        "title": "Match-next-case jump"
```
Comment (voodoos) [inline ocaml-lsp-server/test/e2e-new/code_actions.ml:1344]: If the goal of the new name is to enable keyboard navigation in the code action you should try to have unique first letters like: 
`Match jump`
`Previous case jump`
`Next case jump`
`Let jump`
`Fun jump`

File: ocaml-lsp-server/docs/ocamllsp/config.md (hunk 2 above)
Comment (PizieDust) [reply ocaml-lsp-server/docs/ocamllsp/config.md:51]: thanks for catching that

File: ocaml-lsp-server/test/e2e-new/code_actions.ml (hunk 3 above)
Comment (PizieDust) [reply ocaml-lsp-server/test/e2e-new/code_actions.ml:1344]: will use a function to rename targets that begin with `match-`

**Review by voodoos — APPROVED** [review]

**Review by xvw — APPROVED** [review]

**Comment by voodoos** [comment]
Thanks everyone!

**Pull request merged**